### ISSUE TITLE
Detect wrong uses of `exec` for programs with arguments 

### ISSUE THREAD

**Issue by jjant** [koalaman/shellcheck#2745]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ x I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env

exec "echo with arguments"
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:

`exec` will interpret `"echo with arguments"` as the command name, not `echo` with arguments `with` and `arguments`.
You probably want:

```sh
#!/usr/bin/env

exec "echo" with arguments
```
